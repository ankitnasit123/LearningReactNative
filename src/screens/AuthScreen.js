import {View, Text} from 'react-native';
import React, {useState, useEffect} from 'react';
import InputField from '../components/InputField';

const AuthScreen = () => {
  const [authType, _authType] = useState(0);
  const [showText, setShowText] = useState(true);
  const [userId, setuserId] = useState('');
  const [password, setpassword] = useState('');
  useEffect(() => {
    // Change the state every second or the time given by User.
    const interval = setInterval(() => {
      setShowText(showText => !showText);
    }, 1000);
    return () => clearInterval(interval);
  }, []);

  const handleLogin = () => {
    console.log({userId, password});
  };


  return (
    <>
      <View style={{alignItems: 'center'}}>
        <Text
          style={[
            {
              display: showText ? 'none' : 'flex',
              fontSize: 24,
              color: 'skyblue',
            },
          ]}>
          {authType ? 'Do login. .' : 'Self Register. .'}
        </Text>
        {showText && (
          <Text
            style={[
              {
                fontSize: 24,
                color: 'skyblue',
              },
            ]}>
            {''}
          </Text>
        )}
      </View>
      <InputField
        onChangeText={val => {
          setuserId(val);
        }}
        value={userId}
        secureTextEntry={false}
        placeholder={'Enter user id'}
      />
      <InputField
        onChangeText={val => {
          setpassword(val);
        }}
        value={password}
        secureTextEntry={true}
        placeholder={'Enter pass code'}
      />
    </>
  );
};

export default AuthScreen;
