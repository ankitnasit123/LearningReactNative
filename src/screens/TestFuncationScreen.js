import {Text, View} from 'react-native';
import React, {Component} from 'react';

export default class TestFuncationScreen extends Component {
  constructor(props) {
    super();
    this.state = {
      counter: 0,
    };
  }

  getData(x){
    this.setState({counter: x + 10})
  }
  render() {
    return (
      <View>
        <Text>TestFuncationScreen</Text>
      </View>
    );
  }
}
