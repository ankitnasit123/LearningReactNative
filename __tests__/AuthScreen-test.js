import "react-native";
import React from "react";
import AuthScreen from '../src/screens/AuthScreen'
import TestFuncationScreen from "../src/screens/TestFuncationScreen";

import renderer from "react-test-renderer";

// uncomment if you want to compare with snapshot of code
// test('renders correctly', () => { 
//     const snapshot = renderer.create(<AuthScreen/>).toJSON();
//     console.log(snapshot);
//     expect(snapshot).toMatchSnapshot();
//  })

//in progress with funcational compoent funcation test cases
// test('funcation testing',()=>{
//     const AuthRef = renderer.create(<AuthScreen/>).getInstance();
//     console.log({AuthRef});
//     //const input = AuthRef.handleLogin();
// })

//funcation test with test funcation screen
test('funcation testing',()=>{
    const TestFuncationScreenRef = renderer.create(<TestFuncationScreen/>).getInstance();
    console.log({TestFuncationScreenRef});
    const input = TestFuncationScreenRef?.getData(2);
    console.log({input});
    expect(TestFuncationScreenRef.state.counter).toEqual(12);
})
