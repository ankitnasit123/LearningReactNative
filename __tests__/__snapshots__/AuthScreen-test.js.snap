// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
[
  <View
    style={
      {
        "alignItems": "center",
      }
    }
  >
    <Text
      style={
        [
          {
            "color": "skyblue",
            "display": "none",
            "fontSize": 24,
          },
        ]
      }
    >
      Self Register. .
    </Text>
    <Text
      style={
        [
          {
            "color": "skyblue",
            "fontSize": 24,
          },
        ]
      }
    />
  </View>,
  <TextInput
    onChangeText={[Function]}
    placeholder="Enter user id"
    secureTextEntry={false}
    style={
      {
        "borderRadius": 8,
        "borderStyle": "dashed",
        "borderWidth": 0.5,
        "marginHorizontal": 8,
        "marginVertical": 8,
      }
    }
    value=""
  />,
  <TextInput
    onChangeText={[Function]}
    placeholder="Enter pass code"
    secureTextEntry={true}
    style={
      {
        "borderRadius": 8,
        "borderStyle": "dashed",
        "borderWidth": 0.5,
        "marginHorizontal": 8,
        "marginVertical": 8,
      }
    }
    value=""
  />,
]
`;
